### ISSUE TITLE
devon4j api pom should use spring dependencies

### ISSUE THREAD

**Issue by jdiazgon** [devonfw/devon4j#37]
As the new `devon4j-api` is used to contain api related files like interfaces, I was trying to generate with CobiGen those templates there, so that we are compliant with the new architecture.

One of the templates that I want to generate on the `devon4j-api` project is `SearchCriteriaTo`. This class extends from `AbstractSearchCriteriaTo` which imports `org.springframework.data.domain.Pageable;`.

The problem is that, by default, the pom of `devon4j-api` does not define the spring related dependencies like:
```xml
    <dependency>
      <groupId>com.devonfw.java.starters</groupId>
      <artifactId>devon4j-starter-spring-data-jpa</artifactId>
    </dependency>
```

Therefore we get compilation errors as Pageable is not found.

I would like to have that dependency on the pom by default.

**Comment by hohwille** [comment]
@jdiazgon thanks for this excellent finding.
As the `api` should be minimal from dependencies point of view do not put a dependency on the starter but use `spring-data-commons` instead. Thanks.

**Comment by hohwille** [comment]
Please fix here:
https://github.com/devonfw/devon4j/blob/develop/templates/server/src/main/resources/archetype-resources/api/pom.xml#L15
If you could provide a PR would be awesome.

**Comment by hohwille** [comment]
Merged PR #40, hence done.
